packages/js/src/workouts/components/UnsavedChangesModal.js

```diff
@@ -0,0 +1,161 @@
+
+	/**
+	 * Closes the modal dialog and continues the navigation to the target page.
+	 *
+	 * @returns {void}
+	 */
+	const continueNavigation = useCallback( () => {
+		if ( targetUrl === "popped" ) {
+			window.removeEventListener( "popstate", popStateEventHandler );
+			history.go( -1 );
+		}  else {
+			window.location.replace( targetUrl );
+		}
+	}, [ targetUrl ] );
+
+	/**
+	 * Handles the mouse click event.
+	 *
+	 * @param {Document} event The mouse click event.
+	 *
+	 * @returns {void}
+	 */
+	// eslint-disable-next-line complexity
+	const clickEventHandler = useCallback( ( event ) => {
+		if ( hasUnsavedChanges ) {
+			const adminBarTarget = event.target.closest( ".ab-item" );
+			if ( event.target.tagName === "A" ) {
+				event.preventDefault();
+				window.removeEventListener( "beforeunload", beforeUnloadEventHandler );
+				setTargetUrl( event.target.href );
+				setModalIsOpen( true );
+			} else if ( adminBarTarget ) {
+				if ( adminBarTarget.href && ! adminBarTarget.href.endsWith( "#qm-overview" ) ) {
+					event.preventDefault();
+					window.removeEventListener( "beforeunload", beforeUnloadEventHandler );
+					setTargetUrl( adminBarTarget.href );
+					setModalIsOpen( true );
+				}
+			} else if ( event.target.className === "wp-menu-name" ) {
+				event.preventDefault();
+				window.removeEventListener( "beforeunload", beforeUnloadEventHandler );
+				setTargetUrl( event.target.parentElement.href );
+				setModalIsOpen( true );
+			}
+		}
+	}, [ hasUnsavedChanges ] );
+
+	useEffect( () => {
+		window.addEventListener( "popstate", popStateEventHandler );
+		window.addEventListener( "beforeunload", beforeUnloadEventHandler );
+		window.addEventListener( "click", clickEventHandler );
+
+		return () => {
+			window.removeEventListener( "popstate", popStateEventHandler );
+			window.removeEventListener( "beforeunload", beforeUnloadEventHandler );
+			window.removeEventListener( "click", clickEventHandler );
+		};
+	}, [ beforeUnloadEventHandler, popStateEventHandler, clickEventHandler ] );
+
+	return (
+		<Modal isOpen={ modalIsOpen } handleClose={ closeModal }>
+			<div className="sm:yst-flex sm:yst-items-start">
+				<div
+					className="yst-mx-auto yst-flex-shrink-0 yst-flex yst-items-center yst-justify-center yst-h-12 yst-w-12 yst-rounded-full yst-bg-red-100 sm:yst-mx-0 sm:yst-h-10 sm:yst-w-10"
+				>
+					<ExclamationIcon className="yst-h-6 yst-w-6 yst-text-red-600" aria-hidden="true" />
+				</div>
+				<div className="yst-mt-3 yst-text-center sm:yst-mt-0 sm:yst-ml-4 sm:yst-text-left">
+					<Modal.Title as="h3" className="yst-text-lg yst-leading-6 yst-font-medium yst-text-gray-900">
+						{ __( "Unsaved changes", "wordpress-seo" ) }
+					</Modal.Title>
+					<div className="yst-mt-2">
+						<p className="yst-text-sm yst-text-gray-500">
+							{ __( "There are unsaved changes in this step. Leaving means that those changes will be lost. Are you sure you want to leave this step?", "wordpress-seo" ) }
```

Comment:
[The issue](https://yoast.atlassian.net/browse/DUPP-213) states the change from "page" to "step" should only be done in the first sentence. 
```suggestion
							{ __( "There are unsaved changes in this step. Leaving means that those changes will be lost. Are you sure you want to leave this page?", "wordpress-seo" ) }
```